src/App.js

```diff
@@ -1,19 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-import chatMessages from './data/messages.json';
+import allMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
-  return (
-    <div id="App">
-      <header>
-        <h1>Application title</h1>
-      </header>
-      <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
-      </main>
+    const [chatData, setChatData] = useState(allMessages);
+
+    // pass this function down to ChatEntry using props
+    // updates chatData based on the input
+    const updateChatEntry = (updatedChatEntry) => {
+        const messages = chatData.map((message) => {
```

Comment:
Nice use of map to make the copy of our array so that React will see that the state has changed.